File: app/src/main/java/com/example/domainhunterlite/DomainClassifier.java
package com.example.domainhunterlite;

import com.example.domainhunterlite.utils.RdapResult;

public class DomainClassifier {
    
    public static ClassifiedDomain classify(RdapResult result) {
        DomainType type;
        
        if (isParkedDomain(result)) {
            type = DomainType.PARKED;
        } else if (isEmptyDomain(result)) {
            type = DomainType.EMPTY;
        } else {
            type = DomainType.ACTIVE;
        }
        
        return new ClassifiedDomain(result.domain, type, result.statusCode);
    }
    
    private static boolean isParkedDomain(RdapResult result) {
        String html = result.html.toLowerCase();
        String[] keywords = {
            "domain for sale", "buy this domain", "domain parking",
            "this domain is for sale", "inquiry", "make offer",
            "sedo", "godaddy", "afternic", "dan.com"
        };
        
        for (String keyword : keywords) {
            if (html.contains(keyword)) return true;
        }
        return false;
    }
    
    private static boolean isEmptyDomain(RdapResult result) {
        String html = result.html.toLowerCase();
        
        if (result.statusCode == 404) return true;
        if (html.length() < 5000) return true;
        
        String bodyText = extractBodyText(html);
        return bodyText.length() < 100;
    }
    
    private static String extractBodyText(String html) {
        String text = html.replaceAll("(?i)<script[^>]*>.*?</script>", "");
        text = text.replaceAll("(?i)<style[^>]*>.*?</style>", "");
        text = text.replaceAll("<[^>]+>", "");
        text = text.replaceAll("\\s+", " ");
        return text.trim();
    }
}

File: app/src/main/java/com/example/domainhunterlite/DomainClassifier.kt
package com.example.domainhunterlite

import com.example.domainhunterlite.utils.RdapResult

enum class DomainType {
    EMPTY,      // موقع فارغ
    PARKED,     // للإيجار
    ACTIVE      // يحتوي محتوى
}

data class ClassifiedDomain(
    val domain: String,
    val type: DomainType,
    val statusCode: Int
)

object DomainClassifier {
    
    fun classify(result: RdapResult): ClassifiedDomain {
        val type = when {
            isParkedDomain(result) -> DomainType.PARKED
            isEmptyDomain(result) -> DomainType.EMPTY
            else -> DomainType.ACTIVE
        }
        
        return ClassifiedDomain(
            domain = result.domain,
            type = type,
            statusCode = result.statusCode
        )
    }
    
    private fun isParkedDomain(result: RdapResult): Boolean {
        val html = result.html.lowercase()
        val parkedKeywords = listOf(
            "domain for sale", "buy this domain", "domain parking",
            "this domain is for sale", "inquiry", "make offer",
            "sedo", "godaddy", "afternic", "dan.com"
        )
        
        return parkedKeywords.any { html.contains(it) }
    }
    
    private fun isEmptyDomain(result: RdapResult): Boolean {
        val html = result.html.lowercase()
        
        // صفحة فارغة أو خطأ 404
        if (result.statusCode == 404) return true
        
        // حجم الصفحة صغير جداً (أقل من 5KB)
        if (html.length < 5000) return true
        
        // محتوى فارغ أو قليل جداً
        val bodyText = extractBodyText(html)
        if (bodyText.length < 100) return true
        
        return false
    }
    
    private fun extractBodyText(html: String): String {
        // إزالة script و style
        var text = html.replace(Regex("<script[^>]*>.*?</script>", RegexOption.IGNORE_CASE), "")
        text = text.replace(Regex("<style[^>]*>.*?</style>", RegexOption.IGNORE_CASE), "")
        text = text.replace(Regex("<[^>]+>"), "")
        text = text.replace(Regex("\\s+"), " ")
        return text.trim()
    }
}
